src/execution/mod.rs

```diff
@@ -342,6 +344,260 @@ where
+                                                    run_const(wasm, &mut stack, (), &function_instances);
+                                                    let value = stack.peek_unknown_value();
+                                                    if value.is_none() {
+                                                        panic!("No value on the stack for element segment offset");
+                                                    }
+                                                    value.unwrap()
+                                                };
+
+                                                let offset: u32 = match value {
+                                                    // Value::I32(val) => val,
+                                                    // Value::I64(val) => {
+                                                    //     if val > u32::MAX as u64 {
+                                                    //         panic!("i64 value for data segment offset is out of reach")
+                                                    //     }
+                                                    //     val as u32
+                                                    // }
+                                                    // INFO: no need to implement all of them, it's either i32 or i64, otherwise offset is WRONG
+                                                    // INFO2: wait, we might need globals handling, now that I think about it, but make it return an u32 anyways
+                                                    Value::Ref(rref) => {
+                                                        match rref {
+                                                            Ref::Func(_) => unreachable!(),
+                                                            Ref::Extern(extern_addr) => extern_addr.addr as u32
+                                                        }
+                                                    }
+                                                    _ => {
+                                                        unreachable!()
+                                                    },
+                                                };
+
+                                                Ref::Extern(ExternAddr::new(Some(offset as usize)))
+                                            }).collect::<Vec<Ref>>()
+                                        },
+                                        ElemItems::RefFuncs(_) => panic!("RefFuncs allowed only for Functions!")
+                                    }
+                                }
+                            },
+                        })
+                    },
+                    ElemMode::Active(active_elem) => {
+                        // copies itself right now, when instantiating
+                        let table_idx = active_elem.table as usize;
+                        assert!(tables.len() > table_idx);
```

Comment:
I think assert_debug or an explicit check, to remove panics at runtime

Reply:
This is a tad more complicated. At the moment, when instantiating the store we simply return a Store when, I think, we should return a Result<Store>. The Error should probably be apart of a StoreInstantiationError enum, and thus even the parent (new_with_hooks) return not just a Result<Self, RuntimeError> but a Result<Self, RuntimeError | StoreInstantiationError>

We shall discuss this on Wednesday